[website] Add product core page

siriwatknp opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

waiting for #27666 

**Preview**: https://deploy-preview-27952--material-ui.netlify.app/branding/core/

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

eps1lon commented on docs/src/components/action/Frame.tsx:
```diff
@@ -1,11 +1,10 @@
-import { brandingDarkTheme } from 'docs/src/modules/brandingTheme';
 
-const Demo = (props: BoxProps) => {
+const Demo = React.forwardRef<HTMLDivElement, BoxProps>((props, ref) => {
```
```suggestion
const Demo = React.forwardRef<HTMLDivElement, BoxProps>(function Demo(props, ref) {
```

please name the render functions of `forwardRef` for a proper display name if devtools.

Issue has probably more instances in this PR.

siriwatknp replied:
Got it.

eps1lon reviewed (approved):
Looks great. Just some debugging nit.

oliviertassinari commented:
Looks great, what I could spot:

1. Wrong product name: 
<img width="217" alt="Screenshot 2021-08-30 at 12 12 55" src="https://user-images.githubusercontent.com/3165635/131324040-74ca5375-5b52-4955-a49d-b37d8709fd51.png">
2. Do we need the component function? It feels like we could simplify the demos to be as the inline previews in the documentation.

<img width="254" alt="Screenshot 2021-08-30 at 12 13 48" src="https://user-images.githubusercontent.com/3165635/131324185-804ab4b7-5b53-470b-8d9b-168a22410b1c.png">

3. When the tooltip is selected, should we have one of the 4 tooltips always open? It would avoid having to move the mouse to hover to see the UI. The 3 other tooltips would allow testing the UX.
4. There is an alignment issue with the information icon:

<img width="381" alt="Screenshot 2021-08-30 at 12 17 11" src="https://user-images.githubusercontent.com/3165635/131324566-d3601a97-dd27-4921-ba0c-5de00fa67a97.png">

5. "Take care of your design game" I don't understand this third idea. It seems to duplicate with the first one. How about we use it to demo the responsive API instead?
6. The heading structure feels wrong:

<img width="410" alt="Screenshot 2021-08-30 at 12 21 53" src="https://user-images.githubusercontent.com/3165635/131325159-43bc6273-69ff-4de5-87ae-2952223a5418.png">

First, the most important message seems to be the bigger text, not the smaller one. If it was the case, why would the most important information be with the smaller font-size?

Second, isn't it confusing to have multiple h1 on the page? How do you find the most important piece of information? https://ahrefs.com/blog/h1-tag/

> Use only one H1 per page

https://stripe.com/ might have multiples, it doesn't mean it's right.

7. There is a rogue h6 

<img width="254" alt="Screenshot 2021-08-30 at 12 27 14" src="https://user-images.githubusercontent.com/3165635/131325816-07916742-f8b3-451b-a47a-1f5a7eadbcc7.png">

siriwatknp commented:
@oliviertassinari 

1. fixed
2. agree, updated.
3. agree, updated.
4. nice catch, fixed.
5. good point, change to responsive instead @danilo-leal what do you think about it?
6. changed to `h2, h3` instead but for which one should be `h2`, I am not sure.
7. fixed
8. @danilo-leal any component you want to replace the DatePicker? or should we go with DatePicker for now?

danilo-leal commented:
@siriwatknp @oliviertassinari 

Just pushed a bunch of changes. Regarding bullet nº10, I've removed the date picker and added/tweaked some of the components there. I focused on interactivity instead of a simple display of components, so most of them you can interact and play a bit.

Great work, Jun, looking awesome! The interactive responsive slider on the styling section is just too good, very nice touch!

danilo-leal reviewed (approved):
@siriwatknp @oliviertassinari

Just pushed a bunch of changes. Regarding bullet nº8, I've removed the date picker and added/tweaked some of the components there. I focused on interactivity instead of a simple display of components, so most of them you can interact and play a bit.

Great work, Jun, looking awesome! The interactive responsive slider on the styling section is just too good, very nice touch!

oliviertassinari commented:
> changed to h2, h3 instead but for which one should be h2, I am not sure.

As far as I know, headers are supposed to be continuous h3 + h2 would break this rule. h2 + h3 sounds OK.
